src/centriamTable/CentriamTable.js

```diff
@@ -47,11 +49,22 @@ export default class CentriamTable extends React.Component {
         self.state = {
             data: props.data,
             rowHeight: props.rowHeight,
-            rowClick: props.rowClick  || noop, // noop function
             sortInfo: props.sortInfo || new SortInfo(),
-
+            selectedRow: props.selectedRow
         };
 
+        this.rowClick = rowSelection ? function(data){
```

Comment:
Idea: If you're already keeping track of which row's selected, then you can broadcast another event when the row selection changes. If a row is already selected, you can detect it and not broadcast it.